pkgs/games/openra/mod.nix

```diff
@@ -0,0 +1,86 @@
+# Based on: https://build.opensuse.org/package/show/home:fusion809/openra-ura
+{ stdenv
+, packageAttrs
+, patchEngine
+, mkdirp
+, wrapLaunchGame
+, mod
+, engine
+}:
+
+with stdenv.lib;
+
+stdenv.mkDerivation (recursiveUpdate packageAttrs {
+  name = "openra-${mod.name}-${mod.version}";
+
+  srcs = [
+    mod.src
+    engine.src
+  ];
+
+  sourceRoot = ".";
+
+  postUnpack = ''
+    mv ${engine.src.name} ${mod.src.name}
+    cd ${mod.src.name}
+  '';
+
+  postPatch = ''
+    cat <<'EOF' > fetch-engine.sh
+    #!/bin/sh
+    exit 0
+    EOF
+
+    sed -i 's/^VERSION.*/VERSION = ${mod.version}/g' Makefile
+
+    dos2unix *.md
+
+    ${patchEngine engine.src.name engine.version}
+  '';
+
+  configurePhase = ''
+    make version VERSION=${escapeShellArg mod.version}
+    make -C ${engine.src.name} version VERSION=${escapeShellArg engine.version}
+  '';
+
+  makeFlags = "PREFIX=$(out)";
+
+  checkTarget = "test";
+
+  installPhase = ''
```

Comment:
Add `runHook preInstall`

Reply:
Hmm, I have mixed feelings about this. It is not hard to work around not having these hooks present if you wanted to override something of the package and not all phases/hooks have to be available for a package, like when the `phases` attribute is used, but it is easy enough to add and I checked Nixpkgs and it is used 118 times, so I guess I will just add them.

Reply:
I am still not convinced, because you will have to check the package definition anyway to make sure you are not overwriting any existing `preInstall`. And since only few packages add those `runHook` calls, it is probably better to not assume adding such a phase will just work, instead it is best to always check the package definition. Is there some guideline stating they should be added? Any other arguments to change my mind? I would love to know, so far I have asked twice on the IRC, but gotten no response. However since you seem to feel strongly about it and it does not have any other effect on the build, so I have added them where possible.

Reply:
A naive override of a phase without using `runHook` calls breaks pre- and post- hooks. This is not a consistent interface and can lead to unpleasant user experience. We could say that this is a bug in stdenv and we have to work around it. There is some work on improving this treewide: https://github.com/dezgeg/rfcs/blob/phase-changes/rfcs/0032-run-phase-changes-for-better-nix-shell-use.md

AFAIK there is no guideline for applying this workaround and you are right to point out the lack of consistency in its use across nixpkgs. And I do appreciate that it might make no sense for you to add it to your expression. I think, the only way to really resolve this would be to fix stdenv (e.g by means of RFC above).

Reply:
We expect functions like `buildPythonPackage`, which may override phases, to ensure hooks are still available, but not every individual derivation. It's also difficult to do; what if the derivation implements a `postInstall`, should it then, when explicitly passed in a `postInstall`, override it, or extend it?